Sources/Testing/Issues/Issue+Recording.swift

```diff
@@ -107,6 +107,34 @@ extension Issue {
     let issue = Issue(kind: .errorCaught(error), comments: Array(comment), sourceContext: sourceContext)
     return issue.record()
   }
+  
+  /// Record a new issue when a running test unexpectedly catches an error.
+  ///
+  /// - Parameters:
+  ///   - error: The error that caused the issue.
+  ///   - comment: A comment describing the expectation.
+  ///   - sourceLocation: The source location to which the issue should be
+  ///     attributed.
+  ///   - severity: The severity of the issue.
+  ///
+  /// - Returns: The issue that was recorded.
+  ///
+  /// This function can be used if an unexpected error is caught while running a
+  /// test and it should be treated as a test failure. If an error is thrown
+  /// from a test function, it is automatically recorded as an issue and this
+  /// function does not need to be used.
+  @_spi(Experimental)
+  @discardableResult public static func record(
```

Comment:
You'll want to add a similar overload for the version of `record()` above which accepts a `Comment` as its first parameter. (This one accepts `any Error`.)